### PR TITLE
[WIP] Skeleton for building VStatefulSet controller

### PR DESCRIPTION
**Pull request by codyjrivera**

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -24,7 +25,8 @@ if [ "$registry" != "remote" ] && [ "$registry" != "local" ]; then
     exit 2
 fi
 
-if cd deploy/$1 && kubectl apply -f crd.yaml && kubectl apply -f rbac.yaml && kubectl apply -f deploy_$registry.yaml; then
+## use imperative management for CRDs since metadata for PodTemplateSpec is too long.
+if cd deploy/$1 && kubectl create -f crd.yaml && kubectl apply -f rbac.yaml && kubectl apply -f deploy_$registry.yaml; then
```
Comment (marshtompsxd) [inline deploy.sh:29]: @codyjrivera just curious, why use `create` rather than `apply` here?

File: deploy/vstatefulset/crd.yaml
Comment (marshtompsxd) [inline deploy/vstatefulset/crd.yaml:1]: @codyjrivera can you change every `vstatefulset` to `v_stateful_set`?

File: deploy.sh [hunk 2]
```diff
@@ -14,7 +14,8 @@ NC='\033[0m'
 app=$1
 registry=$2
 
-if [ "$app" != "zookeeper" ] && [ "$app" != "rabbitmq" ] && [ "$app" != "fluent" ]; then
+if [ "$app" != "zookeeper" ] && [ "$app" != "rabbitmq" ] && [ "$app" != "fluent" ]\
+   && [ "$app" != "vstatefulset" ]; then
```
Comment (marshtompsxd) [inline deploy.sh:18]: Can you change `vstatefulset` to `v_stateful_set`?

File: deploy/vstatefulset/vstatefulset.yaml
Comment (marshtompsxd) [inline deploy/vstatefulset/vstatefulset.yaml:1]: Can you change `vstatefulset/vstatefulset.yaml` to `v_stateful_set/v_stateful_set.yaml`?

File: deploy/vstatefulset/crd.yaml
Comment (codyjrivera) [reply deploy/vstatefulset/crd.yaml:1]: Yes.

File: src/vstatefulset_controller.rs
Comment (marshtompsxd) [inline src/vstatefulset_controller.rs:1]: similarly, can you change this to `v_stateful_set_controller.rs`?

File: deploy.sh (hunk 1 above)
Comment (codyjrivera) [reply deploy.sh:29]: The `template: PodTemplateSpec` field of the `VStatefulSet` spec makes the resulting CRD file too large to work with the declarative `kubectl apply`. I admit that using the imperative `kubectl create` is a cheap workaround. Is there an obviously better way of doing this?

**Comment by codyjrivera** [comment]
Stale pull request

**Pull request closed without merging**